syntax = "proto3";

package code_fetcher;

// Service to fetch code from a repository
service CodeFetcher {
  // Fetches code from a given repository URL
  rpc FetchCode (FetchCodeRequest) returns (FetchCodeResponse);
}

// Request message for FetchCode RPC
message FetchCodeRequest {
  string repository_url = 1; // URL of the repository (e.g., GitHub, GitLab)
  // string commit_sha = 2; // Optional: Specific commit to fetch
  // string branch = 3; // Optional: Specific branch to fetch
}

// Response message for FetchCode RPC
message FetchCodeResponse {
  string local_path = 1; // Path to the fetched code on the local filesystem
  Status status = 2; // Status of the fetch operation
  string message = 3; // Optional message (e.g., error details)
}

// Enum for operation status
enum Status {
  UNKNOWN = 0;
  SUCCESS = 1;
  FAILURE = 2;
  IN_PROGRESS = 3;
}